### PR TITLE
Added packer template, vagrant template and answer file for hyper-v g…

### PR DESCRIPTION
**Pull request by johnmccrae**
…eneration 2 VM. Readme also updated to reflect the new work

Signed-off-by: John McCrae <jmccrae@chef.io>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,6 +65,13 @@ $ cd fedora
 $ packer build -var 'mirror=http://mirror.utexas.edu/fedora/linux' fedora-31-x86_64.json
 ```
 
+To build a Windows 10 Enterprise Gen 2 box for only the Hyper-V provider
```
Comment (kagarmoe) [inline README.md:68]:
```suggestion
To build a Windows 10 Enterprise Gen 2 box for the Hyper-V provider
```

File: README.md [hunk 2]
```diff
@@ -65,6 +65,13 @@ $ cd fedora
...
+
+```
+$ cd windows
+$ packer build windows-10gen2.json
+```
+
 If the build is successful, ready to import box files will be in the `builds` directory at the root of the repository.
```
Comment (kagarmoe) [inline README.md:75]: I'm confused about what happens in this sentence. What does "ready to import box files" mean?

File: README.md [hunk 3]
```diff
@@ -89,12 +96,16 @@ Most of the providers expect unrestricted access to networking in order to build
...
-```
+```powershell
 $VS = "Standardswitch"
 $IF_ALIAS = (Get-NetAdapter -Name "vEthernet ($VS)").ifAlias
 New-NetFirewallRule -Displayname "Allow incomming from $VS" -Direction Inbound -InterfaceAlias $IF_ALIAS -Action Allow
 ```
 
+#### Hyper-V Generation 2 VM's
+
+There are fundamental differences between VM types (Gen 1 and Gen 2) in Hyper-V. Most notably, there is no floppy drive support for Gen 2. In order to provide the same resources you previously did with a floppy drive, you will now need to create iso images that contain those files (in particular, the autounattend.xml and the base_setup.ps1 files) and then attach those images via the "secondary_iso_images" parameter. You will also need to manage the driver disk that holds the hyper-v guest services drivers and adjust the autounattend.xml file as appropriate. Also note that there is an updated Autounattend.xml file for Gen 2 which supports EFI partitions. You will want to update that file for the particular Windows version you are using and ensure that the partitions are correct for you situation.
```
Comment (kagarmoe) [inline README.md:107]:
```suggestion
Hyper-V Gen 2 VMs do not support floppy drives. If you previously provided resources using a floppy drive, you must add those files to your Gen 2 iso images, in particular:

- `autounattend.xml`: The Gen 2 `autounattend.xml` file supports EFI partitions. Update the `autounattend.xml` with the correct Windows version for your systems and ensure that the partitions are correct for your situation. You also need to manage the driver disk that holds the hyper-v guest services drivers and adjust the `autounattend.xml` file as appropriate. 
- `base_setup.ps1`

```

Are there any modifications for `base_setup.ps1` in Gen 2?

**Comment by tas50** [comment]
@johnmccrae can you take a look at these docs updates?

File: README.md (hunk 2 above)
Comment (johnmccrae) [reply README.md:75]: I'll fix this in the readme. It should read like "If your build is successful, your box files will be in a builds directory..."

**Pull request merged**